src/manager.cc

```diff
@@ -214,15 +214,23 @@ bool fastcat::Manager::ConfigFromYaml(YAML::Node node)
   return true;
 }
 
-bool fastcat::Manager::Process()
+bool fastcat::Manager::Process(double external_time)
 {
   for (auto it = jsd_map_.begin(); it != jsd_map_.end(); ++it) {
     jsd_read(it->second, 1e6 / target_loop_rate_hz_);
   }
 
   // Pass the PDO read time for consistent timestamping before the device Read()
   //   method is invoked
-  double read_time = jsd_time_get_time_sec();
+  double read_time;
+  if (external_time > -1e9 && external_time < 1e9)
```

Comment:
I recommend adding a parameter in Fastcat's configuration for this.

@alex-brinkman, thoughts?

Reply:
I'm okay with this, it should actually work quite well. 

I'd prefer this, rather than the flavor of logic you have:
```
if (external_time > 0) {
  read_time = external_time;
} else {
  read_time = jsd_time_get_time_sec();
}
```
and then 
```
// src/manager.h
bool Process(double external_time = -1);
```
---
I do have a small fear of someone abusing this on a runtime application (you just never know...) 
So I strongly advocate we prevent this feature when there are any Online JSD devices:

```
if (external_time > 0) {
  if(there_are_any_online_jsd_devices){
    ERROR("Applications cannot use online devices and supply external time, refusing to run");
    return false; 
  }
  read_time = external_time;
} else {
  read_time = jsd_time_get_time_sec();
}
```

Reply:
Any thoughts on a suitable mechanism to check `there_are_any_online_jsd_devices` ?

Reply:
Set a properly named boolean field (initialized to false) true here:
https://github.com/nasa-jpl/fastcat/blob/b4bea35f571305462230c1d174a2946e4080eafd/src/manager.cc#L484

Reply:
Ah just read this right after I did exactly that haha, great minds think alike I guess. Though am realizing I should shunt my boolean back till just before a device is added to the list as you say, lest a rejected device incorrectly flag as a valid online device